azure/services/agentpools/spec.go

```diff
@@ -278,6 +282,7 @@ func (s *AgentPoolSpec) Parameters(existing interface{}) (params interface{}, er
 			EnableAutoScaling:    s.EnableAutoScaling,
 			EnableUltraSSD:       s.EnableUltraSSD,
 			KubeletConfig:        kubeletConfig,
+			KubeletDiskType:      containerservice.KubeletDiskType(to.String((*string)(s.KubeletDiskType))),
```

Comment:
I see this wasn't added to the normalizedProfile used to calculate updates here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2938/files#diff-2c4585415b3fabfd1913f95eb20bc57daac00c2facd3ffa45d3bb00bb1d41b63R189 

My understanding is that it's intentional because the property is immutable but I just want to confirm?

Reply:
I think the link might be pointing to the wrong place, but I don't think this was intentional. I'm not exactly familiar with the normalizedProfile. Would this field be left out since it's immutable?

Reply:
maybe @nojnhuh or @jackfrancis can help answer this

Reply:
GitHub is having trouble rendering that link for me too, but I'm guessing it's here? https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/5b05d6e83e330b16642c121ae515851619615498/azure/services/agentpools/spec.go#L189
@CecileRobertMichon That's not the same field being added in this change, and I see that's conditionally added to `normalizedProfile` below, so that seems fine at first glance to me, albeit slightly asymmetrical with everything else: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/5b05d6e83e330b16642c121ae515851619615498/azure/services/agentpools/spec.go#L208

To your question @willie-yao, yes, if this field is immutable, then we don't need to include it in `existingProfile` or `normalizedProfile`. A diff between those two objects is what ultimately triggers a PUT request to make an update, but the webhook enforces that this field can't change.